pkg/controllers/disruption/emptynodeconsolidation.go

```diff
@@ -102,11 +103,13 @@ func (c *EmptyNodeConsolidation) ComputeCommand(ctx context.Context, disruptionB
 	for _, n := range candidatesToDelete {
 		if len(n.pods) != 0 || c.cluster.IsNodeNominated(n.ProviderID()) || postValidationMapping[n.nodePool.Name] == 0 {
 			logging.FromContext(ctx).Debugf("abandoning empty node consolidation attempt due to pod churn, command is no longer valid, %s", cmd)
-			return Command{}, nil
+			return Command{}, nil, nil
 		}
 		postValidationMapping[n.nodePool.Name]--
 	}
-	return cmd, nil
+	// Return empty scheduling results since no empty nodes should be rescheduling any pods.
+	// Returning nil would require doing a check later in the core controller.
```

Comment:
Having to write this here implies that you now have dependencies on the caller inside of your abstraction. I was trying to think if `scheduling.Results` should be `nil` or not and if we should just handle it gracefully outside....I think it may make sense.

In either case, having a dependency like this between your return arguments seems wrong to me.